src/org/zaproxy/zap/view/table/DefaultCustomColumnHistoryReferencesTableModel.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.zaproxy.zap.view.table;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+
+public class DefaultCustomColumnHistoryReferencesTableModel<T extends DefaultHistoryReferencesTableEntry> extends DefaultHistoryReferencesTableModel  {
```

Comment:
This is not type safe, one can add other types that `T` which end up in `NullPointerException`s later when the values are queried. Since there's other PR adding similar code I'd suggest moving this (and `CustomColumn`) to `callback.ui` package, we can then fix and merge both implementations.